lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw fields to arbitrary names if a mapping is provided
+  if (this.options.fieldMap) {
+    var fieldMap = this.options.fieldMap;
+    results = Utils._.map(results, function(result) {
+      return Utils._.reduce(fieldMap, function(result, name, field) {
+        if (field in result) {
```

Comment:
This depends on if any field name in result is somehow special or not. If it is e.g. `Strange / field` it'll deopt (there was an issue with someone using slashing in his field names :D ). Otherwise it should not but checking for undefined will be faster.